universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -125,19 +126,77 @@ private void putNull(Path path) {
    * Puts the given value at the given path in the underlying JSON data. Builds up the necessary
    * structure along the way, i.e., creates parent objects where necessary.
    *
-   * @param path the {@link Path} with the fully specified path, e.g., "applicant.favorites.color"
-   *     or the equivalent "$.applicant.favorite.colors".
+   * <p>For internal lists
+   *
+   * @param path the {@link Path} with the fully specified path, e.g.,
+   *     "applicant.children[3].favorite_color.text" or the equivalent
+   *     "$.applicant.children[3].favorite_color.text".
    * @param value the value to place; values of type Map will create the equivalent JSON structure
    */
   private void put(Path path, Object value) {
-    if (path.parentPath().isEmpty()) {
-      jsonData.put(Path.JSON_PATH_START_TOKEN, path.keyName(), value);
+    maybePutParent(path);
```

Comment:
suggestion (blocking): I know `maybeDoThing` is an established convention, but it seems it could be improved by making the method name a little longer and saying when it does the thing so the reader doesn't need to look at the implementation or comment to know why it's "maybe". Alternative could be `putParentIfAbsent`

Reply:
done